New branch2

PavelRomashin opened the pull request:
Проверка спринта 9

PavelRomashin committed:
9 спринт

PavelRomashin committed:
9 спринт на проверку

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -2,10 +2,10 @@
+    android:baselineAligned="false"
+    tools:context=".MainActivity">
 
     <FrameLayout
```
Лишняя вложенность, можно обойтись без FrameLayout

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -37,52 +39,25 @@
-            app:iconGravity="textStart"
-            app:iconPadding="@dimen/main_icon_padding"
-            app:iconTint="@color/buttonTextColor" />
+            app:iconTint="@color/backgroundBlack" />
```
Можно цвет иконки задать в самом drawable ресурсе

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -3,28 +3,15 @@
-        android:backgroundTint="@color/white"
-        android:minWidth="@dimen/no_distance"
+        android:id="@+id/backButton"
+        style="@style/settings_header_button"
```
Можно для тулбара использовать androidx.appcompat.widget.Toolbar

faritowich commented on app/src/main/res/values-night/styles.xml:
```diff
@@ -0,0 +1,99 @@
+        <item name="android:orientation">vertical</item>
+    </style>
+
+    <style name="main_button_style" parent="Widget.MaterialComponents.Button">
```
Стиль должен наследоваться от Widget.MaterialComponents.Button.UnelevatedButton.[](url)

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,11 +1,63 @@
+
+        backButton.setOnClickListener {
+            val backIntent = Intent(this, MainActivity::class.java)
+            startActivity(backIntent)
```
Для перехода назад стоит использовать методы finish() или onBackPressed(), потому что сейчас при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на экран настроек

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,11 +1,63 @@
+            supportIntent.putExtra(Intent.EXTRA_EMAIL, email)
+            supportIntent.putExtra(Intent.EXTRA_SUBJECT, subject)
+            supportIntent.putExtra(Intent.EXTRA_TEXT, message)
+            startActivity(supportIntent)
```
Можно ещё использовать котлиновские функции области видимости, чтобы такие фрагменты кода сокращать до таких:

```
            Intent(Intent.ACTION_SENDTO).apply {
                val message = getString(R.string.supportMessage)
                val subject = getString(R.string.supportSubject)
                val email = getString(R.string.email)
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_EMAIL, email)
                putExtra(Intent.EXTRA_SUBJECT, subject)
                putExtra(Intent.EXTRA_TEXT, message)
                startActivity(this)
            }

```
Они иногда удобны, иногда нет, с практикой придёт.
Про такие методы почитать можно здесь:
https://kotlinlang.ru/docs/reference/scope-functions.html

PavelRomashin committed:
Исправление замечаний 1:
1) Убран frame layout
2) Убрано фиксированное значение цвета иконок (правда теперь они прозрачные в конструкторе, но на телефоне отображаются),цвет иконок задан в их xml-файле, почему то цвет из атрибутов не наследуется, задаю его в стиле
3) androidx.appcompat.widget.Toolbar - не нашел его в дизайне, и унаследоваться от него тоже не получается.
4) Наследования стиля в ночном режиме поправил на Widget.MaterialComponents.Button.UnelevatedButton
5) Функцию видимости добавил
6)Метод finish() применен к activity settings и search.
7) Удален режим портрет из манифеста
8) Теперь в окне настроек элементы меню выбираюся по нажатию на строку, а не на иконку
